src/algebra/module/linear_map.lean

```diff
@@ -1,20 +1,41 @@
 /-
 Copyright (c) 2020 Anne Baanen. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Nathaniel Thomas, Jeremy Avigad, Johannes Hölzl, Mario Carneiro, Anne Baanen
+Authors: Nathaniel Thomas, Jeremy Avigad, Johannes Hölzl, Mario Carneiro, Anne Baanen,
+  Frédéric Dupuis, Heather Macbeth
 -/
 import algebra.group.hom
 import algebra.module.basic
 import algebra.group_action_hom
+import algebra.ring.comp_typeclasses
 
 /-!
-# Linear maps
+# (Semi)linear maps
 
 In this file we define
 
-* `linear_map R M M₂`, `M →ₗ[R] M₂` : a linear map between two R-`module`s.
+* `linear_map σ M M₂`, `M →ₛₗ[σ] M₂` : a semilinear map between two `module`s. Here,
+  `σ` is a `ring_hom` from `R` to `R₂` and an `f : M →ₛₗ[σ] M₂` satisfies
+  `f (c • x) = (σ c) • (f x)`. We recover plain linear maps by choosing `σ` to be `ring_hom.id R`.
+  This is denoted by `M →ₗ[R] M₂`.
 
-* `is_linear_map R f` : predicate saying that `f : M → M₂` is a linear map.
+* `is_linear_map R f` : predicate saying that `f : M → M₂` is a linear map. (Note that this
+  was not generalized to semilinear maps.)
+
+## Implementation notes
+
+To ensure that composition works smoothly for semilinear maps, we use the following typeclasses
+  for `ring_hom`s defined in `algebra/ring/comp_typeclasses`:
+* `ring_hom_comp_triple σ₁₂ σ₂₃ σ₁₃`, which states that `σ₁₃ = σ₂₃.comp σ₁₂`.
+* `ring_hom_inv_pair σ σ'`, which state that `σ` and `σ'` are inverses of each other.
+* `ring_hom_surjective σ`, which states that `σ` is surjective.
+These typeclasses ensure that objects such as `σ₂₃.comp σ₁₂` never end up in the type of a
```

Comment:
I think it might be worth repeating this comment in `comp_typeclasses`, and possibly just removing  most of this section here. I'd be inclined to have just

> To ensure that composition works smoothly for semilinear maps, we use the typeclasses `ring_hom_comp_triple`, `ring_hom_inv_pair` and `ring_hom_surjective` from `algebra/ring/comp_typeclasses`.

in this file.

Is it worth pointing out somewhere that a typical use-case would be to use `complex.conj` (or whatever the ring_hom spelling is) as `σ`, so that you get `id` instead of `conj.comp conj`?

Reply:
Good point. I changed it, and included a sentence about conjugate-linear maps as a typical use case.